server/gokbg3/grails-app/controllers/org/gokb/IntegrationController.groovy

```diff
@@ -756,235 +756,278 @@ class IntegrationController {
   @Secured(['ROLE_API', 'IS_AUTHENTICATED_FULLY'])
```

Comment:
We should divide this huge 250 lines method into more intelligible and dealable submethods soon.

Reply:
That is a good idea.